spec/async/barrier_examples.rb

```diff
@@ -0,0 +1,177 @@
+# frozen_string_literal: true
+
+require 'async/clock'
+require 'async/rspec'
+require 'async/semaphore'
```

Comment:
What do you think about zeitwerk?

Reply:
Autoload is broken on Ruby with the fiber scheduler so it's a no go for now, but maybe later?